### PR TITLE
Added Image creation DALL E2 endpoint and model "text-davinci-003" in completion endpoint

### PR DESCRIPTION
**Pull request by PowerRobin**
Added Image creation DALL E2 endpoint and model "text-davinci-003" in completion endpoint

### CODE REVIEW

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin thank you so much for submitting a new update. it looks like the old deployment already started for your connector. we will prepare this update once the current deployment is complete. this should take up to 3-4 weeks. 
Thank you so much.

**Comment by mamurshe** [comment]
@PowerRobin `text-davinci-003` is essentially Chat GPT, is that correct? If yes, can you please include this in the description?

@Amjed-Ayoub how can fast track this deployment?

File: independent-publisher-connectors/OpenAI/apiDefinition.swagger.json [hunk 1]
```diff
@@ -87,20 +87,25 @@
             "name": "engine",
             "in": "path",
             "type": "string",
-            "description": "The used engine, choose between text-davinci-002, text-curie-001, text-babbage-001, text-ada-001",
+            "description": "The used engine, choose between text-davinci-002/003, text-curie-001, text-babbage-001, text-ada-001",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/OpenAI/apiDefinition.swagger.json:90]: @PowerRobin text-davinci-003 is essentially Chat GPT, is that correct? If yes, can you please include this in the description?

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/OpenAI/apiDefinition.swagger.json (hunk 1 above)
Comment (mamurshe) [reply independent-publisher-connectors/OpenAI/apiDefinition.swagger.json:90]: I got a feedback the text-danvinci-003 isn't entirely chat GPT but close. Let's keep it as is an publish. @Amjed-Ayoub

Comment (PowerRobin) [reply independent-publisher-connectors/OpenAI/apiDefinition.swagger.json:90]: ChatGPT is derived from GPT3 text-danvinci-003 model. So basically siblings, but optimized for different things.
Would also like to keep it as is.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by PowerRobin** [comment]
Hey @Amjed-Ayoub something a little bit strange happened.
My last PR was deleted from the repo.

I added "text-davinci-3" model and it actually ended up in production.

It was somehow rolled back and back to the old version right now in production an "text-davinci-3" isn't available anymore.

But this PR is still en route for deployment right? It will reintroduce the model and image generation via DallE2.

I'll probably update the connector soon, because they announced the ChatGPT API. Hope the PR goes through when it is available.

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,

Thank you so much for reaching out. 

You can [move your connector to GA](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-to-ga) if the connector meets our [required criteria](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-to-ga#required-criteria). 
We tested your connector and unfortunately it is not eligible for GA. The connector has no usage at all in the last 7 days, therefore the success rate is 0%. The Connector must have 99.95% availability and the success rate should be greater than an 80% to be eligible for GA. 
I would suggest you make sure that there are some active connections to the connector and that it is being used for a period of time. We can test the connector in 3-4 weeks 😊.

Adding @jopanchal for visibility. 

Thank you so much again for your understanding.

Best regards,
Amjed Ayoub

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,
 
Apologies for the previous message. Thank you so much for reaching out.
Please allow me some time to investigate and I will get back to you as soon as possible.

thank you so much for your patience and understanding.

Best regards,
Amjed Ayoub

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,

I can confirm that we are deploying the latest version for your connector (This PR). The connector is deployed to Asia except Japan and India and to Europe except UK. Please expect the process to complete within this week. 

Thank you so much again.

**Comment by Amjed-Ayoub** [comment]
Hello @PowerRobin,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]

**Pull request merged**